test/pardot-test-srv/main.go

```diff
@@ -0,0 +1,256 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"math/rand/v2"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+)
+
+type config struct {
+	// OAuthAddr is the address (e.g. IP:port) on which the OAuth server will
+	// listen.
+	OAuthAddr string
+
+	// PardotAddr is the address (e.g. IP:port) on which the Pardot server will
+	// listen.
+	PardotAddr string
+
+	// ExpectedClientID is the client ID that the server expects to receive in
+	// requests to the /services/oauth2/token endpoint.
+	ExpectedClientID string `validate:"required"`
+
+	// ExpectedClientSecret is the client secret that the server expects to
+	// receive in requests to the /services/oauth2/token endpoint.
+	ExpectedClientSecret string `validate:"required"`
+
+	// DevelopmentMode is a flag that indicates whether the server is running in
+	// development mode. In development mode, the server will store contacts and
+	// provide an endpoint to query them.
+	//
+	// Only set this flag to true if you are running the server for testing
+	// (e.g. within docker-compose) or local development purposes.
+	DevelopmentMode bool
+}
+
+type token struct {
+	sync.Mutex
+
+	// active is the currently active token. If this field is empty, it means
+	// that the token has been manually expired.
+	active string
```

Comment:
I couldn't find the code that manually expires tokens - where does that happen?

Reply:
Ah, this was removed and I failed to cleanup this remnant of it.